modules/standard/recipe/recipe_controller.py

```diff
@@ -24,134 +24,37 @@ def start(self):
-                        content += "<font color=#FFFFFF><tab>=<tab></font> "
-                        content += "<a href='itemref://%d/%d/%d'>%s</a>" % \
-                                   (result["lowid"], result["highid"], result["ql"],
-                                    self.text.make_image(result["icon"]))
-                        content += "\n<tab><tab>" + self.text.make_item(source["lowid"], source["highid"], source["ql"], source["name"])
-                        content += "\n + <tab>" + self.text.make_item(target["lowid"], target["highid"], target["ql"], target["name"])
-                        content += "\n = <tab>" + self.text.make_item(result["lowid"], result["highid"], result["ql"], result["name"]) + "\n"
-
-                        if "skills" in step:
-                            content += "<font color=#FFFF00>Skills: | %s |</font>\n" % step["skills"]
-                        content += "\n\n"
-
-                    if "details" in recipe:
-                        content += "<font color=#FFFF00>------------------------------</font>\n"
-                        content += "<font color=#FF0000>Details</font>\n"
-                        content += "<font color=#FFFF00>------------------------------</font>\n\n"
-
-                        last_type = ""
-                        for detail in recipe["details"]:
-                            if "item" in detail:
-                                last_type = "item"
-                                i = detail["item"]
-
-                                item = None
-                                if "ql" in i:
-                                    item = self.items_controller.get_by_item_id(i["id"], i["ql"])
-                                else:
-                                    item = self.items_controller.get_by_item_id(i["id"])
-                                    item["ql"] = item["highql"]
-
-                                content += "<font color=#009B00>%s</font>" % \
-                                           self.text.make_item(item["lowid"],
-                                                               item["highid"],
-                                                               item["ql"],
-                                                               item["name"])
-
-                                if "comment" in i:
-                                    content += " - " + i["comment"]
-
-                                content += "\n"
-
-                            elif "text" in detail:
-                                if last_type == "item":
-                                    content += "\n"
-
-                                last_type = "text"
-                                content += "<font color=#FFFFFF>%s</font>\n" % detail["text"]
-
-                    self.db.exec("INSERT INTO recipe (id, name, author, recipe) VALUES (?, ?, ?, ?)", [recipe_id, name, author, content])
+                recipe = self.find_recipe(recipe_id, recipes)
```

Comment:
it might be more efficient to convert the recipes list to a dict with the key as the recipe id, and do lookups that way